File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(string a, string b){
    return a+b>b+a;
}

string solution(vector<int> numbers) {
    string answer = "";
    
    vector<string> str;
    
    for(int i = 0;i<numbers.size();i++){
        str.push_back(to_string(numbers[i]));
    }
    sort(str.begin(),str.end(), compare);
    
    for(int i = 0; i < numbers.size();i++){
        answer+=str[i];
    }
    
    if(str[0] == "0"){
        return "0";
    }
    
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {    
    public String solution(int[] numbers) {
        StringBuilder answer = new StringBuilder();
        int N = numbers.length;
        String[] arr = new String[N];
        for(int i = 0; i < N; i++){
            arr[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(arr, (s1, s2) -> (s2 + s1).compareTo(s1 + s2));
        
        if (arr[0].equals("0")) {
           return "0";
        }
        
        for(int i = 0; i < N; i++){
            answer.append(arr[i]);
        }
        
        String ans = answer.toString();
        
        return ans;
    }
}